source/patterns/@aws-solutions-constructs/aws-lambda-sagemakerendpoint/lib/index.ts

```diff
@@ -0,0 +1,190 @@
+   *
+   * @default - None
+   */
+  readonly existingVpc?: ec2.IVpc;
+  /**
+   * Properties to override default properties if deployVpc is true
+   *
+   * @default - None
+   */
+  readonly vpcProps?: ec2.VpcProps;
+  /**
+   * Whether to deploy a new VPC
+   *
+   * @default - false
+   */
+  readonly deployVpc?: boolean;
+  /**
+   * Whether to deploy a natgatway in the new VPC (if deployVpc is true).
+   * If deployNatGateway is true, the construct creates Public and Private subnets.
+   * Otherwise, it creates Isolated subnets only
+   *
+   * @default - false
+   */
+  readonly deployNatGateway?: boolean;
+  /**
+   * IAM Role, with all required permissions, to be assumed by SageMaker to create resources
+   * The Role is not required if existingSageMakerEndpointObj is provided.
+   *
+   * @default - None
+   */
+  readonly role?: iam.Role;
+}
+
+/**
+ * @summary The LambdaToSageMakerEndpoint class.
+ */
+export class LambdaToSageMakerEndpoint extends cdk.Construct {
+  public readonly lambdaFunction: lambda.Function;
+  public readonly sageMakerEndpoint: sagemaker.CfnEndpoint;
+  public readonly sageMakerEndpointConfig?: sagemaker.CfnEndpointConfig;
+  public readonly sageMakerModel?: sagemaker.CfnModel;
+  public readonly vpc?: ec2.IVpc;
+
+  /**
+   * @summary Constructs a new instance of the LambdaToSageMakerEndpoint class.
+   * @param {cdk.App} scope - represents the scope for all the resources.
+   * @param {string} id - this is a scope-unique id.
+   * @param {LambdaToSageMakerEndpointProps} props - user provided props for the construct.
+   * @since 1.76.0
+   * @access public
+   */
+  constructor(scope: cdk.Construct, id: string, props: LambdaToSageMakerEndpointProps) {
+    super(scope, id);
+
+    if (props.deployVpc || props.existingVpc) {
+      if (props.deployVpc && props.existingVpc) {
+        throw new Error('More than 1 VPC specified in the properties');
+      }
+
+      // If deployNatGateway is true, create Public and Private subnets. Otherwise, create Isolated subnets only
+      const subnetConfiguration: ec2.SubnetConfiguration[] = props.deployNatGateway
+        ? [
+          {
+            cidrMask: 20,
+            name: 'Public',
+            subnetType: ec2.SubnetType.PUBLIC,
+          },
+          {
+            cidrMask: 20,
+            name: 'Private',
+            subnetType: ec2.SubnetType.PRIVATE,
+          },
+        ]
+        : [
+          {
+            cidrMask: 18,
+            name: 'Isolated',
+            subnetType: ec2.SubnetType.ISOLATED,
+          },
+        ];
+
+      // create the VPC
+      this.vpc = defaults.buildVpc(scope, {
+        existingVpc: props.existingVpc,
+        userVpcProps: props.vpcProps,
+        constructVpcProps: {
+          enableDnsHostnames: true,
+          enableDnsSupport: true,
+          // set # NatGateways to 2 if deployNatGateway is true. Otherwise, 0
+          natGateways: props.deployNatGateway ? 2 : 0,
```

Comment:
Why hardcoded 2 here? Are we assuming 2 AZs? CDK programs can configure any number of AZs, isn't reliability usually achieved for NAT Gateways by deploying 1 per AZ used?

From docs: "NAT gateways in each Availability Zone are implemented with redundancy. Create a NAT gateway in each Availability Zone to ensure zone-independent architecture."
https://docs.aws.amazon.com/vpc/latest/userguide/vpc-nat-comparison.html

Reply:
Yes, if `natGateways` is not specified, the CDK will deploy one `natGateway` per AZ by default. The client can reduce the cost by assigning a number, lower than the number of availability zones in the region, to `natGateways` as suggested in the description of `natGateways? ` in [aws-ec2](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.Vpc.html): 
```
...
You can set this number lower than the number of Availability Zones in your VPC in order to save on NAT cost...
```
Because we need to decide whether we want to deploy `natGateways` or not, based on `deployNatGateway?`, we need to assign `0` or `#>0`. 

I've chosen `2` for two reasons:
1. Some regions (e.g. `ca-canada-1`) has only two AZs. 
2. To provide high availability, `natGateways: 1` was not chosen.